Sprint 8-9

Alegiter opened the pull request:
Добавьте в проект темизацию:
* Основа темизации на клиенте, в базе лишь записи userId -> themeId
* Добавлен индекс в таблицу Themes на поле name, чтобы ускорить поиск по имени

Добавьте в проект функционал форума:
* Не сделано 
  > \\3. ответы на комментарии (могут быть рекурсивными и в виде дерева), \\4. эмоции комментариев,

Commit:
Added default Dockerfile

Alegiter committed:
chore(docker): читай README, добавлен docker-compose с postgres + pgAdmin, а также рядом база

Commit:
added app docker config to compose and unsuccessfully tried connect to DB

MankoIvan committed:
branch init

Alegiter committed:
chore(docker): читай обновлённый README в папке docker, донастроен compose нашим приложением

MankoIvan committed:
forum styled

Alegiter committed:
chore: новая проблема в README

Alegiter committed:
Merge branch 'docker_main' into sprint_8

# Conflicts:
#	package.json
#	tsconfig.json

Alegiter committed:
fix(game): убран лишний запрос сохранения рекорда в начале игры

Alegiter committed:
chore(ssr): надо поправить сборку для аудио ассетов

Alegiter committed:
feat(forum): добавлено api + мелкие доработки

Alegiter committed:
feat(themes): добавлено api + мелкие доработки

Alegiter committed:
chore(db): тестовые данные

ivan-infro commented on server.js:
```diff
@@ -1,26 +1,61 @@
+// При запуске из docker обязательно host: postgres; port: 5432
+// При запуске с хостовой машины host: localhost; port: 5555
+const client = new Client({
+  user: 'postgres',
```
Необходимо вынести в переменные окружения

ivan-infro commented on server/database/db-client.ts:
```diff
@@ -0,0 +1,14 @@
+// При запуске из docker обязательно host: postgres; port: 5432
+// При запуске с хостовой машины host: localhost; port: 5555
+const client = new Sequelize({
+  dialect: 'postgres',
```
Должно использовать те же переменные окружения что и код выше

ivan-infro commented on server/index.ts:
```diff
@@ -1,14 +1,42 @@
+app
+  .use(
+    cors({
+      origin: 'https://ya-praktikum.tech',
```
Необходимо вынести в переменные окружения

ivan-infro commented on server/index.ts:
```diff
@@ -1,14 +1,42 @@
+  .use('/api', api)
 // потом всё остальное, что скорее всего запрос за html'кой
   .use(render);
+// todo [sitnik] вариат use('/', render) тоже не помог
```
Почистить todo

ivan-infro commented on server/index.ts:
```diff
@@ -1,14 +1,42 @@
+      await dbClient.sync();
+    })();
+
+    // Start server
```
Нужен ли еще файл src/server.js?

ivan-infro commented on server/middlewares/api/forum/comment.controller.ts:
```diff
@@ -0,0 +1,51 @@
+
+router.post('/', (req, res) => {
+  // todo [sitnik] проверка на application/json
+  const { text, topicId, authorId } = req.body;
```
authorId не должен быть параметром запроса, потому что тогда кто угодно может оставлять комментарии от чужого имени

ivan-infro commented on server/middlewares/api/forum/topic.controller.ts:
```diff
@@ -0,0 +1,52 @@
+  Topic.findByPk(Number(id))
+    .then((topic) => {
+      if (topic) {
+        return topic.countComments().then(() => Promise.resolve(topic));
```
Для чего здесь Promise.resolve?

ivan-infro commented on src/modules/api/UserService.ts:
```diff
@@ -39,4 +39,12 @@ export default class UserService {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    credentials: 'include',
```
Код дублируется в нескольких местах, правильней было бы оформить обертку над fetch, как вы делали это в первом модуле

ivan-infro commented on docker/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+
+COPY  . .
+
+RUN npm run build
```
В package.json отсутствует скрипт build

ivan-infro commented:
Не сохраняется выбранная тема

ivan-infro commented:
Отсутствует проверка авторизации сервере

Alegiter committed:
chore(db): gitkeep для пустых папок базы

Alegiter committed:
fix(db): предсоздание светлой и тёмной темы в базе, если их там нет на начало работы сервера

Alegiter committed:
feat(server): ssl за счёт самоподписного сертификата

Alegiter committed:
feat: закрывание "ручек" за проверкой авторизированности пользователя на сервере

Alegiter committed:
improvement(docker): актуализация Dockerfile под сборку ssr и нового сервера

Alegiter commented:
Тема не сохранялась, потому что я создал две темы в базе руками. И не все папки базы закомитились, из-за чего в спулином проекте нельзя было запустить базу.

Поправил основные ошибки. 
Остальные замечания будут учтены по мере возможности.

Commit:
Added github actions

Alegiter committed:
Merge branch 'forum-api-and-cleanup' into dev

# Conflicts:
#	src/pages/forum/components/topic-previewer/TopicPreviewer.tsx
#	src/pages/forum/pages/topic-list-page/TopicListPage.tsx
#	src/pages/forum/pages/topic-page/components/comment-viewer/CommentViewer.tsx

Alegiter committed:
fix(forum): исправления после мержа + фикс бага с кнопкой "Загрузить ещё"

Alegiter committed:
fix(theme): изменение логики вызова сохранения темы

Alegiter committed:
fix(ssr): серверный StaticRouter работал некорректно

Работал он некорректно из-за отсутствия юзера в сторе. Мидлвар проверки авторизации была перенесена на уровень выше. Из-за чего появилась возможность прокинуть юзера в стор, что исправила роутер.

Alegiter committed:
fix(ssr): дополнительная проверка на случай, если пользователь не авторизован

Alegiter committed:
feat(helmet): добавлен helmet xss+csp

Alegiter committed:
feat: вынесены переменные окружения + донастройки сервера + донастройка билд флоу + убрана база из репы

MankoIvan committed:
Actions autodeploy (#23)

* ssh-action test

* updated project redeploy

* updated project redeploy fix

* updated project redeploy fix 2

* updated project redeploy fix 3

* updated project redeploy fix 4

* ready to push to dev

MankoIvan committed:
fixing white ribbon theme issue

MankoIvan committed:
delete excess test
